src/Aquifer.API/Endpoints/TranslationPairs/Delete/Endpoint.cs

```diff
@@ -0,0 +1,41 @@
+using Aquifer.API.Common;
+using Aquifer.API.Services;
+using Aquifer.Data;
+using FastEndpoints;
+using Microsoft.EntityFrameworkCore;
+
+namespace Aquifer.API.Endpoints.TranslationPairs.Delete;
+
+public class Endpoint(AquiferDbContext dbContext, IUserService userService) : Endpoint<Request>
+{
+    public override void Configure()
+    {
+        Delete("/translation-pairs/{Id}");
+        Permissions(PermissionName.SetTranslationPair);
+    }
+
+    public override async Task HandleAsync(Request req, CancellationToken ct)
+    {
+        var translationPair = await dbContext.TranslationPairs.FirstOrDefaultAsync(x => x.Id == req.Id, ct);
+
+        if (translationPair is null)
+        {
+            await SendNotFoundAsync(ct);
+            return;
+        }
+
+        var user = await userService.GetUserWithCompanyLanguagesFromJwtAsync(ct);
+        var companyLanguage = user.Company.CompanyLanguages.FirstOrDefault(x => x.LanguageId == translationPair.LanguageId);
```

Comment:
Simplification here, you could get rid of the `GetUserWithCompanyLanguagesFromJwtAsync` and use the `GetUserAsync` instead, then:
```
var companyHasLanguagePermission = dbContext.CompanyLanguages.AnyAsync(cl => cl.LanguageId == translationPair.LanguageId && cl.CompanyId == user.CompanyId, ct);
```

Reply:
Wouldn't this be two calls to the database instead of one?